23-input-scanning/exercises/03-unique-words-2/main.go

```diff
@@ -37,4 +45,27 @@ func main() {
 	// [^A-Za-z]+
 	//
 	// Matches to any character but upper case and lower case letters
+
+	var (
+		in          = bufio.NewScanner(os.Stdin)
+		totalUnique int
+		total       int
+		unique      = map[string]bool{}
+	)
+
+	in.Split(bufio.ScanWords)
+	for in.Scan() {
+		regex := regexp.MustCompile("[^A-Za-z]+")
```

Comment:
regex does not change so there is no point in re-creating multiple times. Create it once outside the loop and reuse the same.

Reply:
also name of the regex should tell its purpose like alphaNumRgx ; I know its a little long but it will be more clear

Reply:
Fixed and changed regex to punctRemoveRgx